### PR TITLE
Add expire-at field to clould image metadata

### PR DESCRIPTION
**Pull request by wallyworld**
Add a totally optional expire-at field to cloudimagemetadata.
I'm not revving the version because the field is totally optional - in fact, I have changed import in juju core so that only custom metadata records are imported (those don't have expire-at fields at all). Non-custom records do not need to come across for migration as they will either expire within a few minutes anyway if coming from a 2.3.4 Juju, or needlessly occupy the metadata cache if coming from 2.3.3 or earlier.

### CODE REVIEW

File: cloudimagemetadata.go [hunk 1]
```diff
@@ -211,6 +229,7 @@ func importCloudImageMetadataV1(source map[string]interface{}) (*cloudimagemetad
 		Source_:          valid["source"].(string),
 		Priority_:        int(valid["priority"].(int64)),
 		ImageId_:         valid["image-id"].(string),
+		ExpireAt_:        expireAtPtr,
```
Comment (anastasiamac) [inline cloudimagemetadata.go]: Do we still want to populate it if it does not exist?

Comment (wallyworld) [reply cloudimagemetadata.go]: It will be nil, same as for how root storage size is handled if it is missing.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**